src/bulk_insert/bulk_insert.c

```diff
@@ -7,388 +7,211 @@
 
 #include "bulk_insert.h"
 #include "../stores/store.h"
+#include <errno.h>
 #include <assert.h>
-
-typedef struct {
-    long long node_count;   // Number of nodes to create under this label.
-    const char *label;      // Label.
-    int label_id;           // ID given to label, recognized by graph object.
-    int attribute_count;    // How may attributes are there for this label.
-    char **attributes;      // Array of attributes.
-} LabelDesc;
-
-void _BulkInsert_Reply_With_Syntax_Error(RedisModuleCtx *ctx, const char* err) {
-    RedisModule_ReplyWithError(ctx, err);
+#include <libgen.h>
+#include "../util/rmalloc.h"
+
+// The first byte of each property in the binary stream
+// is used to indicate the type of the subsequent SIValue
+typedef enum {
+    BI_NULL,
+    BI_BOOL,
+    BI_NUMERIC,
+    BI_STRING
+} TYPE;
+
+// Read the header of a data stream to parse its property keys and update schemas.
+static char** _BulkInsert_ReadHeader(GraphContext *gc, LabelStoreType t,
```

Comment:
The very beginning of this function should be a comment explaining the binary format it process